src/utils/filters.js

```diff
@@ -4,7 +4,7 @@
  * @returns Returns all products with the category selected
  */
 export const filterBySubcategory = (products, subcategoryId) => {
-	return products.filter((product) => product.subcategory.id == subcategoryId);
+  return products.filter((product) => product.subcategory.id === subcategoryId);
```

Comment:
I remember that this can cause problems because sometimes `product.subcategory.id` returns an integer, while `subcategoryId` can return a String.

For example, when `subcategoryId` becomes from a router param it is a String, and the comparison `12 === '12'` return `false`, when we are expecting a `true`.